### PR TITLE
feat: enabled configurable server host for the Relay server

### LINKED ISSUE #3072: Enable configurable server host for the Relay server.
### Problem

Currently there's no option to configure server host for the Relay server.

### Solution

- Add new SERVER_HOST to the configuration list
- Enable Relay server to start with configurable host name

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
- Add new SERVER_HOST to the configuration list
- Enable Relay server to start with configurable host name

**Related issue(s)**:

Fixes #3072

**Notes for reviewer**:
Since the server now accepts a configurable `SERVER_HOST`, it will only accept requests from within the same network interface. As a result, in the `Dockerfile`, a new environment variable, `SERVER_HOST`, was introduced. Setting `SERVER_HOST` to `0.0.0.0` in the Docker container's environment allows the container to accept connections from outside, such as from the CI's localhost.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by AlfredoG87 — DISMISSED** [review]
LGTM 👍

**Review by AlfredoG87 — DISMISSED** [review]
LG, nice turn around @quiet-node 👍

**Review by AlfredoG87 — DISMISSED** [review]

**Review by Nana-EC — DISMISSED** [review]

File: packages/server/src/index.ts [hunk 1]
```diff
@@ -22,7 +22,13 @@ import app from './server';
 import { setServerTimeout } from './koaJsonRpc/lib/utils'; // Import the 'setServerTimeout' function from the correct location
 
 async function main() {
-  const server = await app.listen({ port: process.env.SERVER_PORT || 7546 });
+  const port = process.env.SERVER_PORT || 7546;
+
+  // allow `host` to be undefined (same as 0.0.0.0) if not configured
```
Comment (Nana-EC) [inline packages/server/src/index.ts:27]: These comments shuld be part of the configuration md

**Review by Nana-EC — APPROVED** [review]
LG, can move allow host comment to the config doc

File: packages/server/src/index.ts (hunk 1 above)
Comment (quiet-node) [reply packages/server/src/index.ts]: Ah yeah you're right it makes sense. Updated.

**Review by Nana-EC — APPROVED** [review]

**Review by ebadiere — APPROVED** [review]
LG.

**Pull request merged**